Events

Bishoy-Samwel opened the pull request:
- [x] *  Copy the files calculator.js and operate.js into a logic/ directory in your project.
- [x] *  Analyze the files calculator.js and operate.js, be sure you understand what they do.
- [x] *  Import the files in your Calculator component.
- [x] *  use state and props,
- [x] *  pass props to child components,
- [x] *  use lifecycle methods, 
- [x] *  lift state up.
- [x] *  Implement the event handlers you need to use the math logic from calculator.js and operate.js in your React component.
- [x] *  Test your app in the browser. It should be able to perform math operations.

Commit:
Correct divide symbol

Commit:
Add calculate, operate files

Commit:
Change button to p tag and style it

Commit:
Add result component

Commit:
Convert container to a class component

Commit:
Update dependencies

Commit:
handleClick, fix dataObj

Commit:
Add default value for result

mricanho reviewed (changes requested):
Hi @Bishoy-Samwel,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has the Eslint linter that didn't pass the test, remember that all your linters must pass in order to review your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

Commit:
Disable eslint rule

Davidosky007 commented on src/components/Container.js:
```diff
@@ -1,28 +1,55 @@
+
+  render() {
+    const { dataObj } = this.state;
+    console.log(dataObj);
```
* Kindly remove `console.log` from your codebase this is bad practice as important data can be logged in the console and available for anyone to use, we wouldn't want that now, would we?

Davidosky007 commented on src/components/Container.js:
```diff
@@ -1,28 +1,55 @@
+    console.log(dataObj);
+    return (
+      <div className="container">
+        <Result result={dataObj.result} />
```
* I think the result is not updated in the state when i click a button redux-logger is updated but not the result kindly fix this 
* Also i get this error in the console kindly make the necessary adjustments 
![image](https://user-images.githubusercontent.com/52497246/130817305-8d9a097e-8d54-4a5d-8f55-3af32504dbb9.png)

Davidosky007 commented on src/components/Container.js:
```diff
@@ -1,28 +1,55 @@
+    );
+  }
+
+  // this.setState((prevState) => calculate(prevState, buttonName));
```
* I suggest you remove all uncommented code as it makes your codebase bulky and unprofessional

Davidosky007 reviewed (changes requested):
## Hi @Bishoy-Samwel 👋, 

Good job so far! 💪 
* The repo looks professional.✔️
* Git/GitHub flow best practices have been followed✔️
* There are no linter errors✔️
* You have descriptive PR✔️
## There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

Commit:
Fix result type

Commit:
Fix a typo

Commit:
Fix calculate logic

Commit:
Fix calculate for the second time

Commit:
Added rsult to container component state,
Add setstate inside event handler,
Add logic to display the correct result.
Send result state to result component

Commit:
Add componentDidUpdate to watch update

Commit:
Delete  log function

Commit:
Fix linter errors

hemantso reviewed (approved):
Hi @Bishoy-Samwel, 

##Your project is complete! 
There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
